content/blog/2022-08-29-announcing-poetry-1-2-0.md

```diff
@@ -0,0 +1,723 @@
+
+To declare a set of dependencies, which add additional functionality to the project during runtime,
+use [extras]({{< relref "docs/1.2/pyproject#extras" >}}) instead. Extras can be installed by the end user using `pip`.
+
+{{% note %}}
+**A note about the `dev-dependencies` section**
+
+Any dependency declared in the `dev-dependencies` section will automatically be added to a `dev` group.
+So the two following notations are equivalent:
+
+```toml
+[tool.poetry.dev-dependencies]
+pytest = "^7.1.0"
+pytest-mock = "*"
+```
+
+```toml
+[tool.poetry.group.dev.dependencies]
+pytest = "^7.1.0"
+pytest-mock = "*"
+```
+
+Poetry will slowly transition away from the `dev-dependencies` notation which will soon be deprecated, so it's advised
+to migrate your existing development dependencies to the new `group` notation.
+{{% /note %}}
+
+#### Optional groups
+
+A dependency group can be declared as optional. This makes sense when you have a group of dependencies that are only
+required in a particular environment or for a specific purpose.
+
+```toml
+[tool.poetry.group.docs]
+optional = true
+
+[tool.poetry.group.docs.dependencies]
+mkdocs = "*"
+```
+
+Optional groups can be installed in addition to the **default** dependencies by using the `--with` option of
+the [`install`]({{< relref "docs/1.2/cli#install" >}}) command.
+
+```bash
+$ poetry install --with docs
+```
+
+{{% warning %}}
+Optional group dependencies will **still** be resolved alongside other dependencies, so special care should be taken to
+ensure they are compatible with each other.
+{{% /warning %}}
+
+#### Adding a dependency to a group
+
+The [`add`]({{< relref "docs/1.2/cli#add" >}}) command is the preferred way to add dependencies to a group. This is done by
+using the `--group (-G)` option.
+
+```bash
+$ poetry add pytest --group test
+```
+
+If the group does not already exist, it will be created automatically.
+
+#### Installing group dependencies
+
+**By default**, dependencies across **all groups** will be installed when executing `poetry install`.
+
+You can **exclude** one or more groups with the `--without` option:
+
+```bash
+$ poetry install --without docs,test
+```
+
+You can also opt in [optional groups]({{< relref "#optional-groups" >}}) by using the `--with` option:
+
+```bash
+$ poetry install --with docs
+```
+
+If you only want to install the **default**, non-grouped, dependencies, you can do so with the `--only` option:
+
+```bash
+$ poetry install --only main
+```
+
+Finally, if you wish to install **only specific groups** of dependencies without installing the default dependencies,
+the `--only` option can be used like so:
+
+```bash
+$ poetry install --only docs,test
+```
+
+#### Removing dependencies from a group
+
+The [`remove`]({{< relref "docs/1.2/cli#remove" >}}) command supports a `--group` option to remove packages from a specific
+group:
+
+```bash
+$ poetry remove mkdocs --group docs
+```
+
+### New plugin system
```

Comment:
Not sure if we should put it into this section or someone else: We should mention that the `export` command is now provided via a plugin which is installed by default (but we plan to change this in a future version) and it is developed in its own repository with its own issue tracker..

Reply:
Yep, good idea! I won't be near a computer for most of the day, so in case you want to add a commit for that before I'm around, feel free to do so.

Reply:
Done. Feel free to change it.